backend/tests/integration/records/api/test_api_literature.py

```diff
@@ -2263,3 +2263,67 @@ def test_arxiv_url_also_supports_format_alias(inspire_app):
 
     assert response_bibtex.status_code == 200
     assert response_bibtex.content_type == expected_bibtex_type
+
+
+def test_literature_get_modified_authors_after_create(inspire_app):
+    data = {
+        "authors": [
+            {
+                "full_name": "Brian Gross",
+                "ids": [
+                    {"schema": "INSPIRE ID", "value": "INSPIRE-00304313"},
+                    {"schema": "INSPIRE BAI", "value": "J.M.Maldacena.1"},
+                ],
+                "emails": ["test@test.com"],
+            },
+        ]
+    }
+    data = faker.record("lit", with_control_number=True, data=data)
+    record = LiteratureRecord.create(data)
+
+    assert len(list(record.get_modified_authors())) == 1
+
+
+def test_literature_get_modified_authors_after_update(inspire_app):
+    data = {
+        "authors": [
+            {
+                "full_name": "Brian Gross",
+                "ids": [
+                    {"schema": "INSPIRE ID", "value": "INSPIRE-00304313"},
+                    {"schema": "INSPIRE BAI", "value": "J.M.Maldacena.1"},
+                ],
+                "emails": ["test@test.com"],
+            },
+            {
+                "full_name": "Donald Matthews",
+                "ids": [{"schema": "INSPIRE BAI", "value": "H.Khalfoun.1"}],
+                "emails": ["test1@test.pl", "test1.1@test.pl"],
+            },
+        ]
+    }
+    data = faker.record("lit", with_control_number=True, data=data)
+
+    record = LiteratureRecord.create(data)
+
+    data.update(
+        {
+            "authors": [
+                {
+                    "full_name": "Brian Gross",
+                    "ids": [{"schema": "INSPIRE BAI", "value": "B.Gross.1"},],
+                    "emails": ["test@test.com"],
+                    "uuid": "ec485f53-0d75-403a-a11c-e2a6d45dd328",
+                },
+                {
+                    "full_name": "Donald Matthews",
+                    "ids": [{"schema": "INSPIRE BAI", "value": "D.Matthews.1"}],
+                    "emails": ["test1@test.pl", "test1.1@test.pl"],
+                    "uuid": "ec485f53-0d75-403a-a11c-e2a6d45dd327",
+                },
+            ]
+        }
+    )
+
+    record.update(data)
+    assert len(list(record.get_modified_authors())) == 2
```

Comment:
there are two cases causing an author to count as modified:
1. new uuid
2. changed metadata

I think you're only testing 1. here (with very high probability, because UUIDs are random). If you want to test 2. (which would be useful), you need to mutate an author without touching its UUID.